support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -130,26 +132,36 @@ const legend = css`
 	}
 `;
 
-const processPayment = (statusResponse: StatusResponse, geoId: GeoId) => {
-	const { trackingUri, status, failureReason } = statusResponse;
-	const jobId = new URL(trackingUri).searchParams.get('jobId') ?? '';
-	if (status === 'success') {
-		window.location.href = `/${geoId}/thank-you?jobId=${jobId}`;
-	} else if (status === 'failure') {
-		console.error(failureReason);
-	} else {
-		setTimeout(() => {
-			void fetch(trackingUri, {
-				headers: {
-					'Content-Type': 'application/json',
-				},
-			})
-				.then((response) => response.json())
-				.then((json) => {
-					void processPayment(json as StatusResponse, geoId);
-				});
-		}, 1000);
-	}
+/**
+ * This method removes the `pending` state by retrying,
+ * resolving on success or failure only.
+ */
+const processPayment = async (
```

Comment:
This was refactored to just return `success | failure`, removing the `pending` state.

It also leaves the side effects to the calling method.

Reply:
Could not see a `pending` if state in the original tbh, however, I like the way it just returns a failure status object with reason  now.